src/float.cr

```diff
@@ -235,28 +235,28 @@ struct Float64
 
   def hash_normalize
     float_normalize_wrap do
-      {% if flag?(:x86) || flag?(:x86_64) || flag(:arm) || flag(:aarch64) %}
-	# it should work on every architecture where endianess of Float64 and Int64
-	# matches and float is IEEE754.
-	unsafe_int = unsafe_as(Int64)
-	exp = (((unsafe_int >> 52) & 0x7ff) - 1023)
-	mantisa = unsafe_int & ((1_u64 << 52) - 1)
-	if exp > -1023
-	  exp -= 52
-	  mantisa |= 1_u64 << 52
-	else
-	  # subnormals
-	  exp -= 51
-	end
-
-	{% if HASH_BITS == 31 %}
-	  mantisa %= HASH_MODULUS
-	{% end %}
-
-	{mantisa, exp}
-      {% else %}
-	float_normalize_reference
-      {% end %}
+      if FLOAT64_IS_IEEE754
```

Comment:
looks like crystal compiler does very strange things with constants:
even if it knows constant value at compile time and can use it in macros, it still inserts checks for constant's initializer in final code, and doesn't propagate them as real constants to llvm optimizer.

I add `private def hash_bits` and `private def hash_modulus` to workaround this issue.